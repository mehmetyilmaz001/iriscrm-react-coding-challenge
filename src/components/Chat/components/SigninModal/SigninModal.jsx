
import ProptyTypes from 'prop-types';
import { Modal, Box, Typography, Paper, TextField, Button } from '@mui/material';


const style = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 400,
    backgroundColor: 'white',
    // border: '2px solid #000',
    boxShadow: 24,
    p: 4,
    padding: 20
  };

function SigninModal({open, onSubmit}) {
  return (
    <Modal
      open={open}
    //   onClose={handleClose}
      disableEscapeKeyDown={true}
    >
      <Box style={style}>
        <Typography id="modal-modal-title" variant="h6" component="h2">
          Please sign in to continue
        </Typography>

        <Paper 
            component="form" style={{display: 'flex', flexDirection: 'column', gap: 8}} 
            onSubmit={(e) => {
                e.preventDefault();
                const val =  e.target.name.value;
                onSubmit(val);
            }}
        >
            <TextField id="name" label="Name" variant="outlined" />
            <Button type="submit" variant="contained">Sign In</Button>
        </Paper>
        
      </Box>
    </Modal>
  );
}


export default SigninModal;

SigninModal.prototype = {
    open: ProptyTypes.bool,
    onSubmit: ProptyTypes.func
}