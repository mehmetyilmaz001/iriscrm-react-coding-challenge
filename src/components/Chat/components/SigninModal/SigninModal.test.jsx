import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SigninModal from './SigninModal';

describe('SiginModal component', () => {

    const component = mount(<SigninModal />);

    it('expect to render component', () => {
        expect(toJson(component)).toMatchSnapshot();
    });
});