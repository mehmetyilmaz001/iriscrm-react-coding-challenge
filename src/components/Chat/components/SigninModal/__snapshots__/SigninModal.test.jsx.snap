// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiginModal component expect to render component 1`] = `
<SigninModal>
  <ForwardRef(Modal)
    disableEscapeKeyDown={true}
  >
    <ForwardRef(ModalUnstyled)
      BackdropComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "children": [Function],
              "className": [Function],
              "classes": [Function],
              "components": [Function],
              "componentsProps": [Function],
              "invisible": [Function],
              "open": [Function],
              "sx": [Function],
              "transitionDuration": [Function],
            },
            "render": [Function],
          },
          "__emotion_forwardProp": [Function],
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "label:MuiModal-backdrop;",
            Object {
              "zIndex": -1,
            },
            [Function],
            [Function],
          ],
          "defaultProps": undefined,
          "render": [Function],
          "withComponent": [Function],
        }
      }
      closeAfterTransition={false}
      components={
        Object {
          "Root": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": [Function],
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "label:MuiModal-root;",
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "defaultProps": undefined,
            "render": [Function],
            "withComponent": [Function],
          },
        }
      }
      componentsProps={
        Object {
          "root": Object {
            "ownerState": Object {},
          },
        }
      }
      disableAutoFocus={false}
      disableEnforceFocus={false}
      disableEscapeKeyDown={true}
      disablePortal={false}
      disableRestoreFocus={false}
      disableScrollLock={false}
      hideBackdrop={false}
      keepMounted={false}
      onTransitionEnter={[Function]}
      onTransitionExited={[Function]}
    />
  </ForwardRef(Modal)>
</SigninModal>
`;
